pkg/telemetry/publisher/publisher.go

```diff
@@ -92,13 +92,19 @@ func (p *Publisher) publish(ctx context.Context) error {
 		return fmt.Errorf("mark as published: %s", err)
 	}
 
+	sevenDays := 24 * 7 * time.Hour
```

Comment:
Added the logic to delete metrics older than 7 days to avoid the metrics database getting too large. Is 7 days a good number?

Reply:
SGTM